src/main/java/seedu/address/logic/commands/AssignmentGroupCommand.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MAX_SCORE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Assignment;
+import seedu.address.model.person.Group;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+
+/**
+ * Marks the attendance of an existing student in the taa.
+ */
+public class AssignmentGroupCommand extends Command {
+
+    public static final String COMMAND_WORD = "assignGroup";
+    public static final String MESSAGE_SUCCESS = "Assignment created successfully!";
+    public static final String MESSAGE_FAIL = "Assignment not created.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Creates an assignment for students in the same group.\n"
+            + "Compulsory Parameters: "
+            + PREFIX_NAME + "ASSIGNMENT_NAME " + "GROUP " + PREFIX_MAX_SCORE + "MAX_SCORE "
+            + "\nExample: " + COMMAND_WORD + " group1 "
+            + PREFIX_NAME + "Tutorial1 " + PREFIX_MAX_SCORE + "100";
+    private final Name name;
+    private final int maxScore;
+    private final Group group;
+
+    /**
+     * Constructs an AssignmentCommand with the specified name and maximum score.
+     * @param group The group of the students to assign the assignment to.
+     * @param name The name of the assignment. Must not be null.
+     * @param maxScore The maximum possible score for the assignment. Must be greater than 0.
+     * @throws NullPointerException If the provided name is null.
+     * @throws IllegalArgumentException If the provided maxScore is not greater than 0.
+     */
+    public AssignmentGroupCommand(Group group, Name name, int maxScore) {
+        requireNonNull(name);
+        if (maxScore <= 0) {
+            throw new IllegalArgumentException("maxScore must be greater than 0");
+        }
+
+        this.name = name;
+        this.maxScore = maxScore;
+        this.group = group;
+    }
+
+    /**
+     * Executes the command to create a new assignment for all students in the group.
+     * @param model Model representing the current state of the address book.
+     * @return CommandResult representing the result of the execution.
+     * @throws CommandException If there is an error during command execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList().filtered(person -> person.getGroup().equals(group));
```

Comment:
If the tutorial group does not exist, the command will still go through, giving assignments to nobody. Is this expected behaviour?